command/credits.go

```diff
@@ -0,0 +1,248 @@
+		fmt.Fprintln(out, "THANK YOU CONTRIBUTORS!!! <3")
+		fmt.Println()
+	}
+
+	logins := []string{}
+	for x, c := range result {
+		if isTTY && !static {
+			logins = append(logins, getColor(x)(c.Login))
+		} else {
+			fmt.Fprintf(out, "%s\n", c.Login)
+		}
+	}
+
+	if !isTTY || static {
+		return nil
+	}
+
+	lines := []string{}
+
+	thankLines := strings.Split(thankYou, "\n")
+	for x, tl := range thankLines {
+		lines = append(lines, getColor(x)(tl))
+	}
+	lines = append(lines, "")
+	lines = append(lines, logins...)
+	lines = append(lines, "( <3 press ctrl-c to quit <3 )")
+
+	termWidth, termHeight, err := terminal.GetSize(int(outFile.Fd()))
+	if err != nil {
+		return err
+	}
+
+	margin := termWidth / 3
+
+	starLinesLeft := []string{}
+	for x := 0; x < len(lines); x++ {
+		starLinesLeft = append(starLinesLeft, starLine(margin))
+	}
+
+	starLinesRight := []string{}
+	for x := 0; x < len(lines); x++ {
+		lineWidth := termWidth - (margin + len(lines[x]))
+		starLinesRight = append(starLinesRight, starLine(lineWidth))
+	}
+
+	loop := true
+	startx := termHeight - 1
+	li := 0
+
+	for loop {
+		clear()
+		for x := 0; x < termHeight; x++ {
+			if x == startx || startx < 0 {
+				starty := 0
+				if startx < 0 {
+					starty = int(math.Abs(float64(startx)))
+				}
+				for y := starty; y < li+1; y++ {
+					if y >= len(lines) {
+						continue
+					}
+					starLineLeft := starLinesLeft[y]
+					starLinesLeft[y] = twinkle(starLineLeft)
+					starLineRight := starLinesRight[y]
+					starLinesRight[y] = twinkle(starLineRight)
+					fmt.Fprintf(out, "%s %s %s\n", starLineLeft, lines[y], starLineRight)
+				}
+				li += 1
+				x += li
+			} else {
+				fmt.Fprintf(out, "\n")
+			}
+		}
+		if li < len(lines) {
+			startx -= 1
+		}
+		time.Sleep(300 * time.Millisecond)
+	}
+
+	return nil
+}
+
+func starLine(width int) string {
+	line := ""
+	starChance := 0.1
+	for y := 0; y < width; y++ {
+		chance := rand.Float64()
+		if chance <= starChance {
+			charRoll := rand.Float64()
+			switch {
+			case charRoll < 0.3:
+				line += "."
+			case charRoll > 0.3 && charRoll < 0.6:
+				line += "+"
+			default:
+				line += "*"
+			}
+		} else {
+			line += " "
+		}
+	}
+
+	return line
+}
+
+func twinkle(starLine string) string {
+	starLine = strings.ReplaceAll(starLine, ".", "P")
+	starLine = strings.ReplaceAll(starLine, "+", "A")
+	starLine = strings.ReplaceAll(starLine, "*", ".")
+	starLine = strings.ReplaceAll(starLine, "P", "+")
+	starLine = strings.ReplaceAll(starLine, "A", "*")
+	return starLine
+}
+
+func getColor(x int) func(string) string {
+	rainbow := []func(string) string{
+		utils.Magenta,
+		utils.Red,
+		utils.Yellow,
+		utils.Green,
+		utils.Cyan,
+		utils.Blue,
+	}
+
+	ix := x % len(rainbow)
+
+	return rainbow[ix]
+}
+
+func clear() {
+	// on windows we'd do cmd := exec.Command("cmd", "/c", "cls"); unfortunately the draw speed is so
+	// slow that the animation is very jerky, flashy, and painful to look at.
+	cmd := exec.Command("clear")
```

Comment:
Note: if the overhead of shelling out turns out to be too slow for us (with a 300ms refresh rate, that might not yet be the case), you can consider clearing not by running a command, but by sending terminal escape codes

* `fmt.Print("\033c")` - clears the screen (for me, iTerm macOS)
* `fmt.Println("\x1b[%dG")` - since we draw the entire screen every time, we could get away without clearing. this repositions the printing cursor to the start of the terminal

Reply:
ah nice; it has looked fine for me on linux (so i didn't investigate speeding up) but this is great to know